pyOCD/target/cortex_m.py

```diff
@@ -712,17 +712,21 @@ def step(self, disable_interrupts = True):
     def clearDebugCauseBits(self):
         self.writeMemory(DFSR, DFSR_DWTTRAP | DFSR_BKPT | DFSR_HALTED)
 
-    def reset(self, software_reset = False):
+    def reset(self, software_reset = None):
```

Comment:
Internally reset converts None to True.  I set it as None to indicate that no value was specified.  This way calling reset(None) has the same effect as reset().  This allows resetStopOnReset to propagate an unspecified reset value down to reset.